src/components/measurements/measurementsD3.js

```diff
@@ -0,0 +1,338 @@
+import { extent, groups, mean, deviation } from "d3-array";
+import { axisBottom, axisLeft } from "d3-axis";
+import { scaleLinear } from "d3-scale";
+import { select, event as d3event } from "d3-selection";
+import { symbol, symbolDiamond } from "d3-shape";
+import { orderBy } from "lodash";
+
+/* C O N S T A N T S */
+const layout = {
+  leftPadding: 120,
+  rightPadding: 30,
+  topPadding: 20,
+  bottomPadding: 50,
+  subplotHeight: 100,
+  subplotPadding: 10,
+  circleRadius: 3,
+  circleHoverRadius: 5,
+  thresholdStrokeWidth: 2,
+  thresholdStroke: "#DDD",
+  subplotFill: "#adb1b3",
+  subplotFillOpacity: "0.15",
+  diamondSize: 25,
+  standardDeviationStroke: 2,
+  overallMeanColor: "#000"
+};
+
+const classes = {
+  xAxis: "measurementXAxis",
+  yAxis: "measurementYAxis",
+  threshold: "measurementThreshold",
+  subplot: "measurementSubplot",
+  subplotBackground: "measurementSubplotBackground",
+  rawMeasurements: "rawMeasurements",
+  rawMeasurementsGroup: "rawMeasurementsGroup",
+  overallMean: "measurementsOverallMean",
+  colorMean: "measurementsColorMean",
+  mean: "mean",
+  standardDeviation: "standardDeviation"
+};
+
+export const svgContainerDOMId = "measurementsSVGContainer";
+const getMeasurementDOMId = (measurement) => `meaurement_${measurement.measurementId}`;
+const domIdRegex = new RegExp("[^\\w\\-_]+", "gi");
+const getSubplotDOMId = (groupingValue) => `measurment_subplot_${groupingValue.replace(domIdRegex, "_")}`;
+
+/**
+ * Creates the D3 linear scale for the x-axis with the provided measurements'
+ * values as the domain and the panelWidth with hard-coded padding values as
+ * the range. Expected to be shared across all subplots.
+ * @param {number} panelWidth
+ * @param {Array<Object>} measurements
+ * @returns {function}
+ */
+export const createXScale = (panelWidth, measurements) => {
+  return (
+    scaleLinear()
+      .domain(extent(measurements, (m) => m.value))
+      .range([layout.leftPadding, panelWidth - layout.rightPadding])
+      .nice()
+  );
+};
+
+/**
+ * Creates the D3 linear scale for the y-axis of each individual subplot with
+ * the provided measurements' jitters as the domain and the hard-coded
+ * subplot height as the range.
+ * @param {Array<Object>} measurements
+ * @returns {function}
+ */
+export const createYScale = (measurements) => {
+  return (
+    scaleLinear()
+      .domain(extent(measurements, (m) => m.measurementJitter))
+      .range([layout.subplotHeight, 0])
+      .nice()
+  );
+};
+
+/**
+ * Uses D3.groups() to aggregate measurements into a nested array of groups
+ * The groups are sorted with the most number of measurements first.
+ *
+ * If groupByFilters Map is provided, sort the gruops by the order of the keys.
```

Comment:
Minor typo

Reply:
Resolved in https://github.com/nextstrain/auspice/pull/1452/commits/8614feb1339acbf25e9431bbd78f7b1c51ff0273